tests/test_faiss_imputation.py

```
@@ -1,37 +1,145 @@
+
+
+def test_imputer_with_all_nan_column(rng):
+    """Tests if imputer raises error when entire column is NaN"""
+    data = rng.uniform(0, 100, size=(10, 5))
+    data_missing = data.copy()
+    data_missing[:, 2] = np.nan
+    with pytest.raises(ValueError):
+        imputer = FaissImputer(n_neighbors=5)
+        imputer.fit_transform(data_missing)
+
+
+def test_imputer_with_all_nan_row(rng):
+    """Tests if imputer handles all-NaN rows by imputing them"""
+    data = rng.uniform(0, 100, size=(10, 5))
+    data_missing = data.copy()
+    data_missing[3, :] = np.nan
+
+    imputer = FaissImputer(n_neighbors=5)
+    imputed = imputer.fit_transform(data_missing)
+
+    assert not np.isnan(imputed[3, :]).any()
+    assert not np.array_equal(imputed[3, :], data[3, :])
+
+
+def test_imputer_different_n_neighbors(simple_test_df):
+    """Tests if different n_neighbors values produce different results"""
+    data, data_missing = simple_test_df
+    imputer_3 = FaissImputer(n_neighbors=3).fit_transform(data_missing)
+    imputer_7 = FaissImputer(n_neighbors=7).fit_transform(data_missing)
+    assert not np.array_equal(imputer_3, imputer_7)
+
+
+def test_regression_imputation(regression_dataset):
+    """Tests if imputed data maintains predictive power in regression task"""
+    X, X_missing, y = regression_dataset
+    imputer = FaissImputer(n_neighbors=5)
+    X_imputed = imputer.fit_transform(X_missing)
+    assert not np.isnan(X_imputed).any()
+
+    from sklearn.linear_model import LinearRegression
+
+    model_orig = LinearRegression().fit(X, y)
+    model_imputed = LinearRegression().fit(X_imputed, y)
+
+    score_orig = model_orig.score(X, y)
+    score_imputed = model_imputed.score(X_imputed, y)
+    assert abs(score_orig - score_imputed) < 0.1
+
+
+def test_transform_new_data(simple_test_df):
+    """Tests if transform works correctly on new data"""
+    # TODO: Here we have a problem. We can't train with a dataset and impute another one. Test disabled for now
```

We don't really expect our users to do that right? Can we throw an informative error in such cases instead of disabling this test?

---

They can't anymore. Only `fit_transform` is part of the public interface, so they cant call `fit` with a dataset and `transform` with an other. I think the test should be deleted.

---

Let's kill it then :)